idaes/models/costing/SSLW.py

```diff
@@ -1016,13 +1016,32 @@ def cost_compressor(
             integer: whether the number of units should be constrained to be
                 an integer or not (default = True).
         """
-        # Confirm that unit is a turbine
+        # Confirm that unit is a compressor
         if not blk.unit_model.config.compressor:
             raise TypeError(
                 "cost_compressor method is only appropriate for "
                 "pressure changers with the compressor argument "
                 "equal to True."
             )
+        # compressor = True, and using non-isentropic assumption
+        # pumps should use the cost_pump method
+        if (
+            blk.unit_model.config.thermodynamic_assumption
+            == ThermodynamicAssumption.pump
+        ):
+            raise ValueError(
+                f"{blk.unit_model.name} - pressure changers with the pump "
+                "assumption should use the cost_pump method."
+            )
+        # isothermal or adiabatic compressors are too simple to cost
+        elif (
```

Comment:
For future expansion (e.g. if we ever add a polytropic option), it might be better to test for the exceptions instead. I.e.
```
elif blk.unit_model.config.thermodynamic_assumption in [ThermodynamicAssumption.isothermal, ThermodynamicAssumption.adiabatic]: